main.tf

```diff
@@ -238,6 +238,116 @@ resource "aws_cloudwatch_event_connection" "this" {
           password = lookup(basic.value, "password", null)
         }
       }
+
+      dynamic "oauth" {
+        for_each = lookup(each.value.auth_parameters, "oauth", null) != null ? [
+          each.value.auth_parameters.oauth
+        ] : []
+
+        content {
+          authorization_endpoint = lookup(oauth.value, "authorization_endpoint", null)
+          http_method            = lookup(oauth.value, "http_method", null)
+
+          dynamic "client_parameters" {
+            for_each = lookup(each.value.auth_parameters.oauth, "client_parameters", null) != null ? [
+              each.value.auth_parameters.oauth.client_parameters
+            ] : []
+
+            content {
+              client_id     = client_parameters.value.client_id
+              client_secret = client_parameters.value.client_secret
+            }
+          }
+
+          dynamic "oauth_http_parameters" {
+            for_each = lookup(each.value.auth_parameters.oauth, "oauth_http_parameters", null) != null ? [
+              each.value.auth_parameters.oauth.oauth_http_parameters
+            ] : []
+
+            content {
+              dynamic "body" {
+                for_each = lookup(each.value.auth_parameters.oauth.oauth_http_parameters, "body", null) != null ? [
+                  each.value.auth_parameters.oauth.oauth_http_parameters.body
```

Comment:
@antonbabenko could use your help or suggestion here. `body` as well as `header` and `query_string` can occur multiple times. Currently things get override by the last iteration. I was thinking `for b in value : b` would sort it out but then I'm ending up with an unusable value. Perhaps you have an idea?

Reply:
I will take a look later today or tomorrow. I am too busy with other things this week.

Reply:
The magic here is to require a `list of maps` and iterate over them (in S3 and CloudFront modules I did a similar solution, but for 0.13+ the code can be a bit simpler in some cases). I have verified this PR altogether, and it works.

Reply:
@antonbabenko appreciate the help!